### ISSUE TITLE
kubernetes-config: Ability to switch off RBAC resource generation

### ISSUE THREAD

**Issue by edeandrea** [quarkusio/quarkus#32761]
### Description

When using the `kubernetes-config` extension it seems to generate a `RoleBinding` and a `ServiceAccount`. It would be nice to be able to disable this. In some instances when combining with the `kubernetes` extension (and other variants) the user may provide their own `RoleBinding` and/or `ServiceAccount` objects.

This is essentially the same as from #25756 which was implemented for the `kubernetes-client`.

@alexgroom feel free to add any additional detail.

**Comment by alexgroom** [comment]
I found using the OpenShift deployment plugin the build would fail unless you were Project admin (and able to modify RoleBindings and ServiceAccounts), which shouldn't really be assumed. A dev with Project Edit  should still be able to use this feature to update their app. So an OFF switch would be very sensible here

**Comment by geoand** [comment]
Actually, I see that the code also sets a `Role` or a `ClusterRole` in addition to the `ServiceAccount` and `RoleBinding`, so I am not sure if that should be disabled as well...

**Comment by Sgitario** [comment]
> Actually, I see that the code also sets a `Role` or a `ClusterRole` in addition to the `ServiceAccount` and `RoleBinding`, so I am not sure if that should be disabled as well...

At the moment, the Role and ClusterRole generation can be disabled. Though, I think the new property should disable also these two resources as well.

**Comment by geoand** [comment]
> At the moment, the Role and ClusterRole generation can be disabl

But not for `quarkus-kubernetes-config`, or am I missing something?

**Comment by Sgitario** [comment]
> > At the moment, the Role and ClusterRole generation can be disabl
> 
> But not for `quarkus-kubernetes-config`, or am I missing something?

Yes, it can be disabled for the `quarkus-kubernetes-config` extension using the property `quarkus.kubernetes-config.secrets-role-config.generate=false`: https://quarkus.io/version/main/guides/kubernetes-config#quarkus-kubernetes-config_quarkus.kubernetes-config.secrets-role-config.generate

**Comment by geoand** [comment]
Ah okay, yeah, we are talking about the same thing then :).

It seems to me that we need to deprecate the old property and introduce two new ones, one for each `Role` / `ClusterRole` and one for `ServiceAccount` and `RoleBinding`. WDYT?

**Comment by Sgitario** [comment]
> Ah okay, yeah, we are talking about the same thing then :).
> 
> It seems to me that we need to deprecate the old property and introduce two new ones, one for each `Role` / `ClusterRole` and one for `ServiceAccount` and `RoleBinding`. WDYT?

You can only generate either a Role or a ClusterRole at the same time, so we don't need a property for each and I would not also deprecate the existing property, but adding a new property to also control the generation of the `ServiceAccount` and `RoleBinding`. 

The only thing is that the new property would also disable the generation of the Role/ClusterRole resource.

**Comment by geoand** [comment]
> You can only generate either a Role or a ClusterRole at the same time

Right hence the use of `/` :)

**Comment by geoand** [comment]
>  I would not also deprecate the existing property, but adding a new property to also control the generation of the ServiceAccount and RoleBinding

I don't agree, because the current property is called `generate` which means it's generic and therefore it's confusing to only have it control some manifests

**Comment by edeandrea** [comment]
Wouldn't consistency between the `kubernetes-client` & `kubernetes-config` be good?

**Comment by geoand** [comment]
Of course, but does rhe client have the same feature flags?

**Comment by edeandrea** [comment]
Gentle ping on this issue @geoand / @Sgitario / @iocanel

**Comment by edeandrea** [comment]
Another gentle ping on this issue @geoand / @Sgitario / @iocanel ...

**Comment by geoand** [comment]
Given that Jose has moved on to other things and that Ioannis and I are deep into doing other high priority Quarkus related things, I don't see this being picked up any time soon

**Comment by edeandrea** [comment]
I'm happy to work on this if you could give me a gentle nudge to a starting point...

**Comment by geoand** [comment]
I would start [here](https://github.com/quarkusio/quarkus/blob/6f6fb9e331b13b2fe8638e58d633227b9a1a726a/extensions/kubernetes-config/deployment/src/main/java/io/quarkus/kubernetes/config/deployment/KubernetesConfigProcessor.java#L36)
